tsl/src/compression/compression.c

```diff
@@ -1593,6 +1593,14 @@ compress_singlerow(CompressSingleRowState *cr, TupleTableSlot *in_slot)
 	RowCompressor *row_compressor = &cr->row_compressor;
 
 	ExecClearTuple(out_slot);
+	/*
+	 * We need to copy the nulls from the input slot since dropped columns are
+	 * marked as null and `heap_compute_data_size` does not check if a column
+	 * is dropped and only checks if it is null.
+	 */
+	memcpy(out_slot->tts_isnull,
```

Comment:
The out_rel and in-rel have different tuple descriptors. It is not right to blindly copy the in_slot's tts_isnull to out_slot. (You are making assumptions about mapping that could break). Instead, why not init out_slot->tts-null to true and explicitly set out_slot->tts_isnull[#attnum] when it is non-null.

Or set up tts_isnull correctly for dropped columns during out_slot initialization, so that we don't incur this cost every time we copy into the slot.

Reply:
Right, the compressed hypertable have the columns `_ts_meta*` columns. Will fix.

Reply:
Done. I tested to see if it was possible to use the `attisdropped` for the attribute in `in_slot` or `out_slot`, but that does not seem to be set at all (but the null bits are set for the slot).